src/odemis/gui/cont/project.py

```diff
@@ -368,6 +379,7 @@ def __init__(self, number, tab_data, view_ctrl):
 
         self.overlay = None
 
+    @call_in_wx_main  # call in main thread as changes in GUI are triggered
```

Comment:
Indeed, now I see that `canvas.fit_to_bbox()` does use some GUI stuff (while it also tries to not depend on GUI stuff!). So we could add a note in the `fit_to_bbox()` docstring that this function should be called within the main GUI thread.
An alternative to adding the decorator here would be to call `wx.CallAfter(cnvs.fit_to_bbox, [...])` (as it's the last function called, and we don't need the result)

Reply:
I tried the wx.CallAfter (and no decorator) but then I get this error:
`11:37:42 ERROR   main           : Traceback (most recent call last):
  File "/home/sabrina/development/odemis/src/odemis/gui/cont/project.py", line 485, in _on_button
    self.roc_ctrls[num].fit_view_to_bbox()  # Zoom to calibration region
  File "/home/sabrina/development/odemis/src/odemis/gui/cont/project.py", line 399, in fit_view_to_bbox
    wx.CallAfter(cnvs.fit_to_bbox([xmin - 5 * size[0], ymin - 5 * size[1], xmax + 5 * size[0], ymax + 5 * size[1]]))
  File "/usr/lib/python3/dist-packages/wx/core.py", line 3183, in CallAfter
    assert callable(callableObj), "callableObj is not callable"
AssertionError: callableObj is not callable`

Reply:
Please note the syntax. `CallAfter()` takes the function as first argument, and the arguments as following arguments. The way you wrote it immediately called `fit_to_bbox()`, then took the return value (None), and passed it to `CallAfter()`.
You need to write something like `wx.CallAfter(cnvs.fit_to_bbox, [...])`.